kibana-reports/public/components/report_definitions/report_trigger/report_trigger.tsx

```diff
@@ -346,6 +356,24 @@ export function ReportTrigger(props: ReportTriggerProps) {
       };
     }, [cronExpression, timezone]);
 
+    useEffect(() => {
+      if (edit) {
+        httpClientProps
+          .get(`../api/reporting/reportDefinitions/${editDefinitionId}`)
+          .then(async (response) => {
+            if (
+              response.report_definition.trigger.trigger_params
+                .schedule_type === 'Cron based'
+            ) {
+              setCronExpression(
+                response.report_definition.trigger.trigger_params.schedule.cron
+                  .expression
+              );
+            }
+          });
+      }
+    }, []);
+
```

Comment:
why two `useEffect()` are used here

Reply:
The second one is to pre-populate the cron expression if we are on the `Edit` page. Since the dependency array is empty, that `useEffect()` will only execute once. The first `useEffect()` executes whenever either of `[cronExpression, timezone]` are modified, so I separated the Edit behavior into a separate useEffect